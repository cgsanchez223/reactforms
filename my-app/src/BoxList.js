import React, { useState } from 'react';
import Box from './Box';
import NewBoxForm from './NewBoxForm';
import { v4 as uuidv4 } from 'uuid';
import './BoxList.css'

function BoxList({ allBoxes }) {
    const [boxes, setBoxes] = useState(
        allBoxes.map(box => ({ ...box, id: uuidv4() }))
    );

    function addBox(newBox) {
        const boxWithId = { ...newBox, id: uuidv4() };
        setBoxes(boxes => [...boxes, boxWithId]);
    }

    function removeBox(id) {
        setBoxes(boxes => boxes.filter(b => b.id !== id))
    }

    return (
        <div className='boxList'>
            {boxes.map(box => (
                <Box
                    id={box.id}
                    key={box.id}
                    width={box.width}
                    height={box.height}
                    backgroundColor={box.backgroundColor}
                    remove={removeBox}
                />
            ))}
            <NewBoxForm addBox={addBox} />
        </div>
    );
}

BoxList.defaultProps = {
    allBoxes: [
        { width: "50", height: "50", backgroundColor: "blue" },
        { width: "50", height: "50", backgroundColor: "gray" }
    ]
};

export default BoxList;