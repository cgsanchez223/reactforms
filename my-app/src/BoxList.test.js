import React from 'react';
import { render } from '@testing-library/react';
import BoxList from './BoxList';
import { v4 as uuidv4 } from 'uuid';

jest.mock('uuid', () => ({
    v4: jest.fn(() => '1234')
}));

it("renders without crashing", function() {
    render(<BoxList />);
});

it("matches snapshot", function() {
    const { asFragment } = render(<BoxList />);
    expect(asFragment()).toMatchSnapshot();
});