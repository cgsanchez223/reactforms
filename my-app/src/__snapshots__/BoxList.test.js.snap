// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="boxList"
  >
    <div
      data-testid="box-undefined"
    >
      <div
        class="box"
        style="width: 50px; height: 50px; background-color: blue;"
      />
      <button>
        X
      </button>
    </div>
    <div
      data-testid="box-undefined"
    >
      <div
        class="box"
        style="width: 50px; height: 50px; background-color: gray;"
      />
      <button>
        X
      </button>
    </div>
    <form>
      <h1>
        Add new box
      </h1>
      <div
        class="formGroup"
      >
        <label
          for="width"
        >
          Width: 
        </label>
        <input
          id="width"
          name="width"
          placeholder="Enter width"
          type="text"
          value=""
        />
      </div>
      <div
        class="formGroup"
      >
        <label
          for="height"
        >
          Height: 
        </label>
        <input
          id="height"
          name="height"
          placeholder="Enter height"
          type="text"
          value=""
        />
      </div>
      <div
        class="formGroup"
      >
        <label
          for="backgroundColor"
        >
          Background Color: 
        </label>
        <input
          id="backgroundColor"
          name="backgroundColor"
          placeholder="Enter background color"
          type="text"
          value=""
        />
      </div>
      <button>
        Add Item
      </button>
    </form>
  </div>
</DocumentFragment>
`;
